src/contracts/entrypoints.rs

```diff
@@ -387,14 +387,25 @@ pub trait InvariantTrait {
     ///
     /// # Parameters
     /// - `pool_key`: A unique key that identifies the specified pool.
-    /// - `offset`: The offset from which ticks will be retrieved.
+    /// - `tick_indexes`: Indexes of the tick to be retrieved.
     #[ink(message)]
-    fn get_liquidity_ticks(&self, pool_key: PoolKey, offset: u16) -> Vec<LiquidityTick>;
-
+    fn get_liquidity_ticks(
+        &self,
+        pool_key: PoolKey,
+        tick_indexes: Vec<i32>,
+    ) -> Result<Vec<LiquidityTick>, InvariantError>;
     /// Retrieves the amount of liquidity ticks of a specified pool.
     ///
     /// # Parameters
     /// - `pool_key`: A unique key that identifies the specified pool.
+    /// - `lower_tick`: index to start counting from
+    /// - `upper_tick`: index to stop counting after(inclusive)
+    ///
     #[ink(message)]
-    fn get_liquidity_ticks_amount(&self, pool_key: PoolKey) -> u32;
+    fn get_liquidity_ticks_amount(
```

Comment:
As I understand, there is always enough gas for this entrypoint, even in the worst-case scenario where the tick spacing is equal to 1 and the maximum search range is used? It's always worth including such information in the docs

Reply:
I find this in SDK `// Query needs to be split in the case where tickSpacing = 1, otherwise a single query will fit within the gas limit`